### PR TITLE
Update libcalico-go for new CRDs

### PR DESCRIPTION
**Pull request by caseydavenport**

### CODE REVIEW

File: tests/mock_data/kdd/ipam.yaml [hunk 1]
```diff
@@ -6,6 +6,8 @@ metadata:
 spec:
   node: kube-master
   cidr: 192.168.221.192/26
+  deleted: ""
```
Comment (neiljerram) [inline tests/mock_data/kdd/ipam.yaml:9]: @caseydavenport Am I right that these `deleted: ""` additions should be reverted if your proposed libcalico change (to make `deleted` an option boolean) is merged?

Comment (caseydavenport) [reply tests/mock_data/kdd/ipam.yaml:9]: @neiljerram Yeah, if we change the type then we'll need to update these lines to `deleted: false`.

The issue is that we were applying yamls here that weren't fully qualified, and the schema mandates that the fields are present. That will be the same even with my PR, independent of the type of the field.

Comment (neiljerram) [reply tests/mock_data/kdd/ipam.yaml:9]: Even with the `+optional` tag?

Comment (caseydavenport) [reply tests/mock_data/kdd/ipam.yaml:9]: Yeah, we could add that and `omitempty` to make it an optional field, and then remove these lines, but I wasn't planning on doing that. It's an internal API and all of the code does actually write these fields out today, so seems like we should keep doing that unless there's a reason.

**Pull request merged**